src/main/java/org/creekservice/kafka/test/perf/implementations/SkemaImplementation.java

```diff
@@ -71,22 +78,34 @@ public JsonValidator prepare(
             final AdditionalSchemas additionalSchemas,
             final boolean enableFormatAssertions) {
 
-        /*
-        Waiting on https://github.com/erosb/json-sKema/commit/ee8aca8c1452688dba485fa7cc7f3ab4fc85d74c being released
-        before we can make use of enableFormatAssertions
-         */
-
         final JsonValue schemaJson = new JsonParser(schema).parse();
 
+        final SchemaClient schemaClient =
+                new SchemaClient() {
+                    @NotNull
+                    @Override
+                    public InputStream get(@NotNull final URI uri) {
+                        return new ByteArrayInputStream(
+                                additionalSchemas.load(uri).getBytes(UTF_8));
+                    }
+
+                    @NotNull
+                    @Override
+                    public IJsonValue getParsed(@NotNull final URI uri) {
+                        return SchemaClient.DefaultImpls.getParsed(this, uri);
+                    }
+                };
+
         final SchemaLoader schemaLoader =
-                new SchemaLoader(
-                        schemaJson,
-                        new SchemaLoaderConfig(
-                                uri ->
-                                        new ByteArrayInputStream(
-                                                additionalSchemas.load(uri).getBytes(UTF_8))));
-
-        final Validator validator = Validator.forSchema(schemaLoader.load());
+                new SchemaLoader(schemaJson, new SchemaLoaderConfig(schemaClient, "mem://input"));
+
+        final Validator validator =
+                Validator.create(
+                        schemaLoader.load(),
+                        new ValidatorConfig(
+                                enableFormatAssertions
+                                        ? FormatValidationPolicy.ALWAYS
+                                        : FormatValidationPolicy.NEVER));
```

Comment:
Or maybe this should be `DEPENDS_ON_VOCABULARY`?

Reply:
See: https://github.com/creek-service/json-schema-validation-comparison/pull/122#issuecomment-1906988565